### ISSUE TITLE
Extract Class to read video from file

### ISSUE THREAD

**Issue by jat113** [TJenks13/StreamVideo-1#17]
Considering performing an extract class to encapsulate reading document video.csv. This way we can create  videos separately from loading the document. An advantage of this is that the function that creates the video does not need to know how the file was read. A disadvantage is that data binding the document creates new objects so it does take more memory.

I think it would be useful however.

**Comment by henry-nnnn** [comment]
So make a class for LoadVideo?/Turn the vector array into an object/class?

**Comment by jat113** [comment]
I already did the refactor and will commit to a branch, but I can't push it until the test cases are written for video_t which I'm waiting for that to be finished and pushed. But what I did was created a function createRecord which the csv file and puts each field (video title, video type,etc.) into a class I called DataRecord. Then when we want to load a video, we pass in a DataRecord object and that creates the video. 

In this way the function that reads the file knows nothing about creating a video. And the function that creates a video knows nothing about how to read the file. I think that's a worthwhile encapsulation even if it does add an extra class.

**Comment by jat113** [comment]
I just pushed the changes in the branch issue8. You can see what I did there if I wasn't clear. As I said, I'd push it but going to let Trevor finish the unit test cases for video_t before I do.

**Comment by jat113** [comment]
I pulled in my changes.  I saw that Trevor had already completed the test cases. Just never pulled it.

**Comment by jat113** [comment]
Looks like we have test cases for all the files now so we can do whatever refactors we desire on any files.

**Comment by henry-nnnn** [comment]
my god, josh, save us some work too haha

**Comment by jat113** [comment]
I've noticed many more refactors we can perform still so I don't think we need to worry about running out. For example, I don't like the way I'm constructing streams in the function CreateStream in StreamingActivity. Similarly, I dislike the way I'm constructing the type of the video in the constructor for the video class using a DataRecord object. I think the factory pattern would be great for both cases, but I'm not sure. 

Also, there's a single if statement in Account report that checks if the type of stream is an OriginalStream. I really hate that if statement and it seems like a definite indication that I did not replace the initial switch statement with polymorphism as well as I could have. 

I also think that entire function report in Account has a ton of parameters. I think that in the video class we can delegate a structure like RunningTime that will contain the hours and minutes of the video and that would be a useful refactor. In general there's a ton of primitives floating around in the code and your issue you created for the video class brought that to my attention. 

So there's a lot we can still refactor.
